Mage.Sets/src/mage/cards/t/TreeOfRedemption.java

```diff
@@ -78,7 +75,7 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
         player.setLife(amount, game, source);
-        game.addEffect(new SetBasePowerToughnessSourceEffect(null, StaticValue.get(life), Duration.WhileOnBattlefield, SubLayer.SetPT_7b), source);
+        game.addEffect(new SetBaseToughnessSourceEffect(life, Duration.Custom), source);
```

Comment:
So why exactly Duration.Custom instead of Duration.WhileOnBattlefield?
From my understanding when activating it multiple time, there are just stacking continuous layer on the same layer, timestamp are taking care of having the last one be the base toughness, and all of them are cleared when the Tree leaves the battlefield.

Reply:
(I still don't totally understand, but correct usage seems to be WhileOnBattlefield for static abilities and Custom for abilities that get added.)